integrations/watsonx/src/haystack_integrations/components/generators/watsonx/chat/chat_generator.py

```diff
@@ -0,0 +1,391 @@
+        :param api_base_url: Custom base URL for the API endpoint.
+            Defaults to "https://us-south.ml.cloud.ibm.com".
+        :param generation_kwargs: Additional parameters to control text generation.
+            These parameters are passed directly to the watsonx.ai inference endpoint.
+            Supported parameters include:
+            - `temperature`: Controls randomness (lower = more deterministic)
+            - `max_new_tokens`: Maximum number of tokens to generate
+            - `min_new_tokens`: Minimum number of tokens to generate
+            - `top_p`: Nucleus sampling probability threshold
+            - `top_k`: Number of highest probability tokens to consider
+            - `repetition_penalty`: Penalty for repeated tokens
+            - `length_penalty`: Penalty based on output length
+            - `stop_sequences`: List of sequences where generation should stop
+            - `random_seed`: Seed for reproducible results
+        :param timeout: Timeout in seconds for API requests.
+            Defaults to environment variable `WATSONX_TIMEOUT` or 30 seconds.
+        :param max_retries: Maximum number of retry attempts for failed requests.
+            Defaults to environment variable `WATSONX_MAX_RETRIES` or 5.
+        :param tools: List of tools in Watsonx format for function calling.
+            Each tool should be a dictionary with:
+            - `name`: Tool name
+            - `description`: Tool description
+            - `parameters`: JSON schema for parameters
+        :param verify: SSL verification setting. Can be:
+            - True: Verify SSL certificates (default)
+            - False: Skip verification (insecure)
+            - Path to CA bundle for custom certificates
+        """
+        self.api_key = api_key or Secret.from_env_var("WATSONX_API_KEY")
+        self.model = model
+        self.project_id = project_id or Secret.from_env_var("WATSONX_PROJECT_ID")
+        self.api_base_url = api_base_url or "https://us-south.ml.cloud.ibm.com"
+        self.generation_kwargs = generation_kwargs or {}
+        self.timeout = timeout or float(os.environ.get("WATSONX_TIMEOUT", "30.0"))
+        self.max_retries = max_retries or int(os.environ.get("WATSONX_MAX_RETRIES", "5"))
+        self.tools = tools
+        self.verify = verify
+
+        if not project_id:
+            msg = "project_id must be provided"
+            raise ValueError(msg)
+
+        self._initialize_client()
+
+    def _initialize_client(self):
+        """Initialize the Watsonx client with configured credentials."""
+        credentials = Credentials(api_key=self.api_key.resolve_value(), url=self.api_base_url)
+
+        self.client = ModelInference(
+            model_id=self.model,
+            credentials=credentials,
+            project_id=self.project_id.resolve_value() if self.project_id else None,
+            verify=self.verify,
+            max_retries=self.max_retries,
+            delay_time=0.5,
```

Comment:
I couldn't see any source documentation on this variable. Could you explain what `delay_time` does and why it should be set to `0.5` versus the default of `None`?